kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/cache/ReflectorWatcher.java

```diff
@@ -48,7 +48,7 @@ public void eventReceived(Action action, T resource) {
       log.error(errorMessage);
       throw new KubernetesClientException(errorMessage);
     }
-    log.info("Event received {}", action.name());
+    log.debug("Event received {}", action.name());
```

Comment:
I think trace might be even more appropriate here as a lot of people run with debug level output in development and will still get spammed.

Reply:
@hanikesn @rohanKanojia I've added another commit taking this down to trace and rationalizing the other logs.

Reply:
and a further change to just combine the logs

Reply:
I'm sorry, not following what happened here. The resulting changes include a log.debug and not a log.trace.

Reply:
I've updated the title of the pull request.  After the request above to use trace, instead of debug, I took some more looks at what was going on with logging in general in ReflectorWatcher.  Now I'm proposing to combine two logs in eventReceived at the debug level along with other logging related changes.

Reply:
With the latest change, I'm assuming that resource cannot be null even with an error action - because that assumption was already being made in the logging of the error action https://github.com/fabric8io/kubernetes-client/pull/2993/files#diff-5c9996a3493c8638b0e14ee9a8f95a4e6efd8a83e966de0013a4f1bd8c97807fR55

However it appears that it can be null from https://github.com/fabric8io/kubernetes-client/blob/74cc63df9b6333d083ee24a6ff3455eaad0a6da8/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java#L108

What is the actual expectation?

Reply:
I think earlier we had the wrong expectation of resource coming not null in case of error event:
https://github.com/fabric8io/kubernetes-client/blob/f8bfc5b9189af6cbbb997a2a61c036d4db8aa249/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/cache/ReflectorWatcher.java#L54

From looking at code, seems like resource is always `null` in case of `Action.ERROR`:
```
kubernetes-client : $ git grep -in "Action.Error,"
kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java:108:          manager.eventReceived(Action.ERROR, null);
kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchHTTPManager.java:241:      manager.eventReceived(Action.ERROR, null);
kubernetes-client/src/test/java/io/fabric8/kubernetes/client/dsl/base/WaitForConditionWatcherTest.java:115:    watcher.eventReceived(Action.ERROR, configMap);
```

I think it's better to remove it or just log error event received.

Reply:
@rohanKanojia I've updated to account for null resources